src/mono/mono/metadata/gc.c

```diff
@@ -908,7 +908,7 @@ static void
 init_finalizer_thread (void)
 {
 	ERROR_DECL (error);
-	gc_thread = mono_thread_create_internal ((MonoThreadStart)finalizer_thread, NULL, MONO_THREAD_CREATE_FLAGS_NONE, error);
+	gc_thread = mono_thread_create_internal ((MonoThreadStart)finalizer_thread, NULL, MONO_THREAD_CREATE_FLAGS_DEFER_PLATFORM_INIT, error);
```

Comment:
So should we call `mono_thread_init_platform_state` inside `finalizer_thread` somewhere?

Reply:
Oops, good catch. Yes, the Finalizer thread should call that. I don't know about Mono but in CoreCLR the Finalizer thread lives the process lifetime so technically it will not drain until the process terminates.

Reply:
Yep, Mono's too.

Reply:
Found a nasty bug adding this. I was using `mono_runtime_try_invoke_handle` instead of `mono_runtime_try_invoke_handle_void` and saw many asserts when I tried initializing the platform.